src/replica/split/replica_split_manager.cpp

```diff
@@ -1172,5 +1172,87 @@ void replica_split_manager::trigger_secondary_parent_split(
     // TODO(heyuchen): add other split_status check, response will be used in future
 }
 
+// ThreadPool: THREAD_POOL_REPLICATION
+void replica_split_manager::copy_mutation(mutation_ptr &mu) // on parent partition
+{
+    dassert_replica(_child_gpid.get_app_id() > 0, "child_gpid({}) is invalid", _child_gpid);
+
+    // TODO(hyc): if copy mutation synchronously, add flags
+
+    mutation_ptr new_mu = mutation::copy_no_reply(mu);
+    error_code ec = _stub->split_replica_exec(
+        LPC_PARTITION_SPLIT,
+        _child_gpid,
+        std::bind(&replica_split_manager::on_copy_mutation, std::placeholders::_1, new_mu));
+    if (ec != ERR_OK) {
+        parent_cleanup_split_context();
+    }
+}
+
+// ThreadPool: THREAD_POOL_REPLICATION
+void replica_split_manager::on_copy_mutation(mutation_ptr &mu) // on child partition
+{
+    if (status() != partition_status::PS_PARTITION_SPLIT) {
+        derror_replica(
+            "wrong status({}), ignore this mutation({})", enum_to_string(status()), mu->name());
+        _stub->split_replica_error_handler(
+            _replica->_split_states.parent_gpid, [mu](replica_split_manager *split_mgr) {
+                split_mgr->parent_cleanup_split_context();
+                split_mgr->on_copy_mutation_reply(
+                    ERR_OK, mu->data.header.ballot, mu->data.header.decree);
+            });
+        return;
+    }
+
+    if (!_replica->_split_states.is_prepare_list_copied) {
```

Comment:
```suggestion
    if (!_replica->_split_states.is_prepare_list_copied) {
```
What does this statement mean? If prepare-list hasn't copied, why don't we proceed with copying? Why return?

Reply:
When parent partition set valid child_gpid, it will be considered doing partition split, child partition should copy each mutation. At the same time, child partition will be created, and its status is PS_PARTITION_SPLIT, parent will prepare states for child including prepare list. Parent partition and child partition may not execute in the same thread, it is possible for parent partition start to send mutation but child partition has not copied parent prepare list. In this case, child partition should not copy mutation.